SDL/Event.cu

```diff
@@ -2162,7 +2140,7 @@ SDL::segments* SDL::Event::getSegments()
         segmentsInCPU->phi = new float[N_MAX_PIXEL_SEGMENTS_PER_MODULE];
         segmentsInCPU->seedIdx = new unsigned int[N_MAX_PIXEL_SEGMENTS_PER_MODULE];
         segmentsInCPU->isDup = new bool[N_MAX_PIXEL_SEGMENTS_PER_MODULE];
-        segmentsInCPU->isQuad = new bool[N_MAX_PIXEL_SEGMENTS_PER_MODULE];
+        segmentsInCPU->isQuad = new short[N_MAX_PIXEL_SEGMENTS_PER_MODULE];
```

Comment:
Why the change of type here (and also below)? `bool` seems to make more sense, no?

Reply:
This was causing a bug in the code. If you notice in master, the isQuad variable is created in the sdl.cc code as a vector of type short. Then after it gets moved through this kernel it gets manually copied to an array on the GPU which has type bool. So if you just try to replace the current manual copy with an automatic copy the data types don't match and it isn't copied correctly. The solution is to either change everything to type short or type bool. Initially I tried changing everything to bool type but it gave me an error saying I couldn't use the pointer of a "temporary rvalue" when copying data to the GPU. If you change everything to data type short though it works fine.

Let me try changing everything to bool again and see if I can get it to work. I agree that bool type makes more sense.

Reply:
Just to log this, I was not able to get the bool datatype working properly. As a better solution, I changed the datatype from short to char everywhere. Since char and bool have the same memory footprint this is a slight improvement over master which has isQuad as a short initially.

Reply:
For documentation, here is a link with pointers to the reason why this doesn't work and an alternative way to deal with it:
https://stackoverflow.com/questions/39701660/fast-way-of-conversion-of-stl-vectorbool-to-bool-array